Projmgr documentation (#205)
Open-CMSIS-Pack/devtools: tools/toolbox/docs/installation.md

  - [Linux x86_64 Tarball](https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2)
  - [Mac OS X 64-bit Package](https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-mac.pkg)

- [**Keil MDK**](http://www.keil.com/mdk5/install) version 5.36 or higher 

- [**ARM Compiler Version 6**](https://developer.arm.com/tools-and-software/embedded/arm-compiler/downloads/version-6) **license managed**:

  Version 6.16 (Mar. 10th 2020)
  - [Windows 32-bit Installer](https://developer.arm.com/-/media/Files/downloads/compiler/DS500-BN-00025-r5p0-18rel0.zip)

    - Download installer
    - Extract archive unzip DS500-BN-00025-r5p0-18rel0.zip
    - Run win-x86_32\setup.exe
    - Default installation path: C:\Program Files (x86)\ARMCompiler6.16\

  - [Linux x86_64 Installer](https://developer.arm.com/-/media/Files/downloads/compiler/DS500-BN-00026-r5p0-18rel0.tgz)
    - Download installer
    - Extract the archive tar -xzf DS500-BN-00026-r5p0-18rel0.tgz
    - Run install_x86_64.sh

## Installation

>Note: Before installation, ensure that the required tools listed above are installed.

> To install the CMSIS-Toolbox a bash environment is required. install for example [git for Windows](https://gitforwindows.org). Call from the bash prompt:

Reviewer: ```suggestion
To install the CMSIS-Toolbox a bash environment is required. Install for example [git for Windows](https://gitforwindows.org). Call from the bash prompt:
```

```

Below the manual setup of the CMSIS-Toolbox is explained. The command below installs the tools into the directory `./ctools`.
```txt
./cmsis-toolbox.sh -x ./ctools
```

## Configuration

For the manual installation described above the configuration setup is as follows:

### ./etc/setup

In file `<cmsis-toolbox-installation-dir>/etc/setup` the paths to the central pack directory (`CMSIS_PACK_ROOT`), the compiler definition files (`CMSIS_COMPILER_ROOT`), and the binaries (`CMSIS_BUILD_ROOT`) are defined.

```txt
export CMSIS_PACK_ROOT=/C/Keil/ARM/PACK
export CMSIS_COMPILER_ROOT=/c/ctools/etc
export CMSIS_BUILD_ROOT=/c/ctools/bin
```

#### Git Bash Setup

In a bash environment the following command configures the environment:

```txt
source <cmsis-toolbox-installation-dir>/etc/setup
```

#### Windows Command Line Setup

For Windows use the **System Properties** dialog and add the following **Environment Variables**:

Variable             | Value
:--------------------|:---------------
CMSIS_PACK_ROOT      | Path to the central CMSIS-Pack folder (i.e. C:\Keil\ARM\PACK)
CMSIS_COMPILER_ROOT  | Path to the CMSIS-Toolbox 'etc' directory (i.e. C:\ctools\etc)

> Path                 | Add the path to the CMSIS-Toolbox 'bin' directory (i.e. C:\ctools\bin)

Reviewer: ```suggestion
Path                 | Add to the system path the CMSIS-Toolbox 'bin' directory (i.e. C:\ctools\bin)
```

### ./etc/setup/\*.cmake

> The supported of the various toolchains is defined by `*.cmake` files in the directory `<cmsis-toolbox-installation-dir>/etc/setup`. The filenames have the following format:

Reviewer: ```suggestion
The support of the various toolchains is defined by `*.cmake` files in the directory `<cmsis-toolbox-installation-dir>/etc/setup`. The filenames have the following format:
```

> Note: The filenames reflect the available compiler versions on the host system.  There may be multiple files for each compiler to support different versions, for example `AC6.6.16.0.cmake` and `AC6.6.18.0.cmake`.

> Each of this `*.cmake` files define the path (`TOOLCHAIN_ROOT`) to the toolchain binaries, the file extension (`EXT`) of the executable binaries, and other compiler related parameters for the invocation. Edit the files to reflect the path as shown in the example (for `AC6`) below:

Reviewer: ```suggestion
Each of these `*.cmake` files defines the path (`TOOLCHAIN_ROOT`) to the toolchain binaries, the file extension (`EXT`) of the executable binaries, and other compiler related parameters for the invocation. Edit the files to reflect the path as shown in the example (for `AC6`) below:
```

```txt
############### EDIT BELOW ###############
# Set base directory of toolchain
set(TOOLCHAIN_ROOT "C:/Keil/ARM/ARMCLANG/bin")
set(EXT .exe)
```

## Using VS Code 

VS Code is an effective environment to create CMSIS-based projects.  As [**csolution**](../../projmgr/docs/Manual/Overview.md) files are in YAML format, it is recommended to install:

- [**YAML Language Support by Red Hat**](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).

To work with the **CMSIS-Toolbox** in VS Code use:

> - **Terminal - New Terminal** to open a terminal window, you may chose as profile `bash`, `powershell`, or `Command prompt`.

Reviewer: ```suggestion
- **Terminal - New Terminal** to open a terminal window, you may choose as profile `bash`, `powershell`, or `Command prompt`.
```
